js/partial.ts

```diff
@@ -0,0 +1,37 @@
+import { Scorer, ScorerArgs } from "@braintrust/core";
+
+export interface ScorerWithPartial<Output, Extra>
+  extends Scorer<Output, Extra> {
+  partial: <T extends Partial<ScorerArgs<Output, Extra>>>(
```

Comment:
What do you think about something like this?

```
partial: <T extends keyof Extra>(args: { [K in T]: Extra[K] }) => Scorer<Output, Omit<Extra, T> & Partial<Pick<Extra, T>>>
```

Here, `args` only permits sub-members of `Extra` which match exactly the type signature in `Extra`. So we cannot provide arguments which don't exist in `Extra`, nor can we provide `undefined` for arguments which are required in `Extra`, and thus launder an `undefined` into the actually-invoked function.

Also, this feels like a nice general utility. Something like:
```
function partial<F extends (args: any) => any, PartialArgs extends keyof Parameters<F>[0]>(
    f: F,
    partialArgs: { [K in PartialArgs]: Parameters<F>[0][K] }
) {
    function ret(args: Omit<Parameters<F>[0], PartialArgs> & Partial<Pick<Parameters<F>[0], PartialArgs>>) {
        return f({
            ...partialArgs,
            ...args,
        });
    }
    // Do any function metadata adjustment too.
    return ret;
}
```

Reply:
Nice. The new type signature works for me. I'll leave the generic utility function out for now.